core/src/main/java/io/fabric8/maven/core/config/MappingConfig.java

```diff
@@ -0,0 +1,24 @@
+package io.fabric8.maven.core.config;
+
+import org.apache.maven.plugins.annotations.Parameter;
+
+public class MappingConfig {
+
+    @Parameter(required = true)
+    private String kind;
+
+    @Parameter(required = true)
+    private String filenames;
```

Comment:
Maybe `filenameKinds` (see https://maven.fabric8.io/#kind-filename-type-mapping) ? And just remove the @Parameter annotation.